### ISSUE TITLE
Can you stream multiple arrow tables back ?

### ISSUE THREAD

**Issue by erikseulean** [apache/arrow#40853]
I have a usecase where in one do_get request I would like to stream back multiple arrow tables. Currently the stream generator accepts only one schema and a stream. 

Assuming that I want to run two sql queries (this is a simplified example) that return two arrow tables, that potentially have different schemas, how do I stream both of them back to the client ? 

Is this supported ?

### Component(s)

FlightRPC, Python

**Comment by zeroshade** [comment]
Currently, the spec would require you to send a separate DoGet request for each query and have it return the two streams separately.

In theory you *could* leverage the multiple `FlightEndpoints` that are returned to be different streams (one stream per table) but that is technically against the current Flight spec which assumes that multiple flight endpoints are pieces of the same result set, and thus should have the same schema. So if you did this it wouldn't necessarily be compatible with many flight clients. 

So the best options currently are:

1. Use separate DoGet requests that each return their own result stream
2. Propose a change to the Flight spec to allow multiple different data streams identified back from the same FlightInfo object
3. Revive the discussion on allowing schema changes / evolution within a single IPC stream

**Comment by erikseulean** [comment]
Thanks for your reply. 

I think there is value in being able to return multiple tables that have different schemas for efficiency reasons. There are cases where you might load two different sql tables that have completely different schemas as it is more efficient than sending two DoGet requests.

**Comment by zeroshade** [comment]
I agree that there's value in these, I highly encourage you to write something up and send an email to the dev@arrow.apache.org mailing list. These will require changes in semantics of the protocols which will need to be agreed on and prototyped by the community to be integrated into the server/client implementations.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 365 days. Please remove the stale label or comment below, or this issue will be closed in 14 days. If this usage question has evolved into a feature request or docs update, please remove the 'Type: usage' label and add the 'Type: enhancement' label instead.
